### PR TITLE
Fix beacon category disappear bug

### LINKED ISSUE #389: Categories disappearing from Beacon opportunities after publishing
Why?!

### LINKED ISSUE #401: Add test for contract type filter in Scout

### PR DESCRIPTION
**Pull request by bsmithgall**
### What Changed
- Fixed bug where categories disappear on Beacon
- Added tests for category updating
- Added test for contract type filter on scout
### Issues
- Fixes #389 
- Fixes #401

### CODE REVIEW

File: purchasing/opportunities/admin/views.py [hunk 1]
```diff
@@ -68,7 +64,7 @@ def edit(opportunity_id):
             form, categories, subcategories = generate_opportunity_form(obj=opportunity)
 
             if form.validate_on_submit():
-                form_data = fix_form_categories(request, form, Opportunity, opportunity)
+                form_data = fix_form_categories(request, form, Opportunity, obj=opportunity)
```
Comment (bsmithgall) [inline purchasing/opportunities/admin/views.py:67]: This is the fix @hamhands pretty sad I know

Comment (hamhands) [inline purchasing/opportunities/admin/views.py:67]: Hacky hacker is never sad. It's not pretty, but w/e.

**Comment by hamhands** [comment]
I approve of this hack. LGTM :+1:

**Comment by hamhands** [comment]
Re-posted the ems billing rfp and it emailed immediately. So that's cool.

**Pull request merged**